### ISSUE TITLE
Disconnect with large queries

### ISSUE THREAD

**Issue by BenGu3** [prisma/prisma#15168]
### Bug description

When running a `findMany` query with parameters such as `findMany({ where: { OR: orQuery } })` where `orQuery` is too large, Prisma throws a `Can't reach database server at` error. 

This repository includes code and instructions to reproduce this error:
https://github.com/BenGu3/prisma-db-disconnect

### How to reproduce

Follow instructions in the following repository:
https://github.com/BenGu3/prisma-db-disconnect

### Expected behavior

Either the query runs as long as it needs to, or Prisma throws a more precise error explaining that the query is too large to process.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}


model User {
  id    String @id @default(uuid()) @db.Uuid
  email String @unique

  sourceRequests FriendRequest[] @relation("sourceUser")
  targetRequests FriendRequest[] @relation("targetUser")
}

model FriendRequest {
  sourceId String @db.Uuid
  targetId String @db.Uuid

  sourceUser User @relation("sourceUser", fields: [sourceId], references: [id])
  targetUser User @relation("targetUser", fields: [targetId], references: [id])

  @@id([sourceId, targetId])
}
```

```ts
async function main() {
  const userCount = 200
  const fetchCount = 17000

  console.log(`Creating ${userCount} users....`)
  const users = times(userCount, () => ({ id: faker.datatype.uuid(), email: faker.internet.email() }))
  await prisma.user.createMany({ data: users })
  console.log('Done creating users.')
  console.log()

  console.log(`Creating ${userCount * userCount} friend requests...`)
  const friendRequests = users.flatMap(sourceUser => {
    return users.map(targetUser => ({
      sourceId: sourceUser.id,
      targetId: targetUser.id,
    }))
  })
  await prisma.friendRequest.createMany({ data: friendRequests })
  console.log('Done creating friend requests.')
  console.log()

  console.log(`Fetching ${fetchCount} friend requests...`)
  const res = await prisma.friendRequest.findMany({ where: { OR: friendRequests.slice(0, fetchCount) }})
  console.log(`Done fetching ${res.length} friend requests.`)
}
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: 16.15.0

### Prisma Version

```
prisma                  : 4.3.1
@prisma/client          : 4.3.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/prisma-fmt-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.3.0-32.c875e43600dfe042452e0b868f7a48b817b9640b
Default Engines Hash    : c875e43600dfe042452e0b868f7a48b817b9640b
Studio                  : 0.473.0

```

**Comment by jkomyno** [comment]
Thank you for this issue. I'm closing it as a duplicate in favour of https://github.com/prisma/prisma/issues/8832.

We also have a reproduction [here](https://github.com/jkomyno/reprod-8832) :)

**Comment by jkomyno** [comment]
We have closed the original issue in `prisma@4.4.0`, please see [this comment](https://github.com/prisma/prisma/issues/8832#issuecomment-1260533913).
